back/engines/commercial/insights/app/services/insights/front_end_format_text_network.rb

```diff
@@ -8,83 +8,62 @@ module Insights
   # This class implements a representation for text networks that is convenient
   # to work with for the front-end.
   class FrontEndFormatTextNetwork
-    DEFAULT_KEYWORD_SIZE_RANGE = [1, 5].freeze
-    DEFAULT_CLUSTER_SIZE_RANGE = [100, 500].freeze
+    DEFAULT_NODE_SIZE_RANGE = [1, 5].freeze
     MAX_NB_CLUSTERS = 10
-    MAX_NB_KW_PER_CLUSTER = 25 # max number of keyword by cluster
+    MAX_NB_KEYWORDS = 100
+    MAX_DENSITY = 0.05
 
     attr_reader :id
 
     # @param [Insights::View] view
     def initialize(
       view,
-      keyword_size_range: DEFAULT_KEYWORD_SIZE_RANGE,
-      cluster_size_range: DEFAULT_CLUSTER_SIZE_RANGE,
-      max_nb_clusters: MAX_NB_CLUSTERS,
-      max_nb_kw_per_cluster: MAX_NB_KW_PER_CLUSTER
+      node_size_range: DEFAULT_NODE_SIZE_RANGE,
+      max_nb_nodes: MAX_NB_KEYWORDS,
+      max_density: MAX_DENSITY
     )
       @id = "network-#{view.id}"
-      @keyword_size_range = keyword_size_range
-      @cluster_size_range = cluster_size_range
+      @node_size_range = node_size_range
+      @max_density = max_density
 
       @network = NLP::TextNetwork.merge(
         # Namespacing networks wrt to the language to avoid id collisions.
         *view.text_networks.map { |tn| tn.network.namespace(tn.language) }
       )
-      @network.prune_communities(max_nb_clusters)
-      @network.shrink_communities(max_nb_kw_per_cluster)
+
+      @network.remove_orphan_nodes
+      @network.prune_communities(MAX_NB_CLUSTERS)
```

Comment:
Yeah, I can get that 😅 I removed it from parameters because each cluster is associated to one colour and we're using a palette with 10 colours. If there are more than 10 clusters, the FE starts looping over the colours, which made sense with the previous visualization (you could still tell clusters apart, even when they had the same colour), but that's not the case anymore with the flat network.

Actually, now I think about it, it is maybe not the best decision as dealing with colours is the FE's responsibility.